pkg/node/node.go

```diff
@@ -283,11 +292,20 @@ func NewNode(
 		)
 
 		// Create a new node manager to keep track of compute nodes connecting
-		// to the network.
+		// to the network. Provide it with a mechanism to lookup (and enhance)
+		// node info, and a reference to the heartbeat server if running NATS.
 		nodeManager := manager.NewNodeManager(manager.NodeManagerParams{
-			NodeInfo: tracingInfoStore,
+			NodeInfo:   tracingInfoStore,
+			Heartbeats: heartbeatSvr,
 		})
 
+		// Start the nodemanager, ensuring it doesn't block the main thread and
+		// that any errors are logged. If we are unable to start the manager
+		// then we should not start the node.
+		if err := nodeManager.Start(ctx); err != nil {
+			return nil, errors.Wrap(err, "failed to start node manager")
+		}
+
```

Comment:
Any reason not to perform this construction in the `NewRequesterNode` method?

Reply:
heartbeatSvr is constructed from a client that is optionally created in NewNode().  It should probably move as part of rework/refactor how these nodes are constructed as at present there's been attempt to isolate all the libp2p/nats stuff in NewNode()